chia/util/errors.py

```diff
@@ -309,3 +312,14 @@ class InvalidPathError(Exception):
     def __init__(self, path: Path, error_message: str):
         super().__init__(f"{error_message}: {str(path)!r}")
         self.path = path
+
+
+class ServerError(Exception):
+    pass
+
+
+class SamePeerIdError(ServerError):
+    def __init__(self, peer: PeerInfo, node_id: bytes32) -> None:
+        super().__init__(f"Trying to connect to a ourselves: {peer} {node_id}")
```

Comment:
I guess it would make more sense to not talk about connection in here since `SamePeerIdError` sounds more generic than just "we connect to ourself". Also storing `peer` and `node_id` without any usage seems pointless? :) Not sure what else i would do but this seems not well thought out so far? Maybe just don't add a `__init__` for `SamePeerIdError` and instead pass in the message in where you raise it?

Reply:
The wording of the message is wrong.  The rest seems useful?  Taking parameters and formatting the message brings consistency.  Storing the parameters makes them available when someone wants to catch the exception and be aware of the relevant information.  Yes, people could just add it when they need it.  Makes for bigger diffs and this also leads people to have exceptions which can be handled better I think.

I'll revisit the message and see what you think of the rest.

Reply:
> Storing the parameters makes them available when someone wants to catch the exception and be aware of the relevant information. 

_When_ someone wants to use it. Can you imagine a case where its useful for this exception? I just don't think its very helpful to add not used things all over the place only because they _might_ be used in the future but most likely will just stay there unused forever.. 

> Yes, people could just add it when they need it. Makes for bigger diffs and this also leads people to have exceptions which can be handled better I think.

Either now or then the diff is bigger. If it gets added when its actually used you can at least better understand why it was introduced when checking the history, i think.